common/prque/sstack.go

```diff
@@ -81,7 +115,38 @@ func (s *sstack) Len() int {
 // Compares the priority of two elements of the stack (higher is first).
 // Required by sort.Interface.
 func (s *sstack) Less(i, j int) bool {
-	return (s.blocks[i/blockSize][i%blockSize].priority - s.blocks[j/blockSize][j%blockSize].priority) > 0
+	var result bool
+
+	switch s.priorityType {
+	case Int:
+		i := (s.blocks[i/blockSize][i%blockSize].priority).(int)
```

Comment:
`s.blocks[i/blockSize][i%blockSize].priority` and `s.blocks[j/blockSize][j%blockSize].priority` are too repetitive. How about assigning them to variables at first?

Reply:
It is reflected in #840 PTAL :)